### ISSUE TITLE
Xperia Pro/Mini Pro keyboard "world key" not supported

### ISSUE THREAD

**Issue by peti** [LegacyXperia/local_manifests#504]
First of all, thanks a lot for your work on this project. The LegacyXperia project is awesome! Thank you.

Now, I installed the latest CM10.2 ROMS on my Xperia Pro, which was manufactured for the German market and hence comes with a QWERTZ layout. A picture of the keyboard can be seen at http://www-static.se-mc.com/blogs.dir/0/files/2011/10/xperia-pro-silver-sideview-android-smartphone-940x529.png.

The keyboard works fine except for the "world key" that you can see on the right hand-side, directly above the "cursor right" key. In the stock Android version, it was possible to use that key to switch between multiple configured languages. (The soft keyboard does have that key, too, and there it works exactly like this.) Furthermore, it was possible the press that key for a longer while and open up the "Configure input method" dialog.

With CyanogenMod 10.2 installed, however, the key acts like caps-lock key, and neither one of those two features works.

It's not a big issue, but IMHO it would be nice if the original behavior would be (optionally?) available.

**Comment by piotrwie** [comment]
I've remapped the key to act as ctrl. this is quite useful as some apps support ctrl+key combinations. 
for example browser:
ctrl+f - search, ctrl+t new browser tab, ctrl+w - close browser tab, ctrl+l - immediately pops the address bar, sometimes even combos ctrl+c, vtrl+v work, etc. 
you need to edit /system/usr/key-layout/pmxxxx....kl file and give it a line:
KEY 368   CTRL_LEFT      WAKE_DROPPED
instead of default meaning (just replace word capslock (or whatever there was) with ctrl_left for the 358 key).

when i broke the file it switched to some default layout which is basically standard qwerty but without the alt combos.
you need to remount system partition for this to read-write.
then reboot.

Piotr

**Comment by piotrwie** [comment]
of course i mean 368 not 358 in description above.

**Comment by A-n-D** [comment]
this issue is not much bothering, but i'm facing a broblem with function and caps key in secured https sites... after single or double pressing key doesn't work(as it works in normal http links and in stock rom), i have to press and hold function(or caps) key n the while holding it press another key for entering any icon or entering capital lettr.  can anybody confirm this?

**Comment by piotrwie** [comment]
Yes, I can confirm it. On some web pages the fn and shift do not keep their supposed function, so you need to keep shift pressed while pressing other keys. Mini pro, released 10.2.

A-n-D notifications@github.com napisał(a):

**Comment by A-n-D** [comment]
(mk16i)pro cm11,  cm 10.1, cm 10.2, same behaviour in all released... 
can u create an issue on bug tracker... (eng is not my first language so i fear, i won't b able to describe problem correctly if i made an issue, but i feel this problem need to get fix)

**Comment by mikeNG** [comment]
I was donated an xperia pro so I might be able to take a look at this finally
